
<!DOCTYPE html PUBLIC>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="templates/templatetibomain.css" />
<title>Thibault Leroy - Genomics & Evolution</title>
</head>

<body>
    <div id="page">
        <div class="topNaviagationLink"><a href="index.html">Home</a></div>
        <div class="topNaviagationLink"><a href="courses.html">Courses</a></div>
        <div class="topNaviagationLink"><a href="publications.html">Publications & CV</a></div>
	    <div class="topNaviagationLink"><a href="https://github.com/ThibaultLeroyFr" target="_blank">Scripts & Pipelines</a></div>
	<div class="topNaviagationLink"><a href="index2.html">Contact</a></div>
	</div>
    <div id="mainPicture">
    	<div class="picture">
        	<div id="headerTitle">Thibault Leroy</div>
            <div id="headerSubtext">Evolutionary biologist</div>
        </div>
    </div>
        <div class="contentBox">
    	<div class="innerBox">

		
	<h1>Courses</h1>
	<h2>Population genomic approaches (Master & Phd students)</h2>      
		

	<h3>      -Importance of premiminary analyses in population genomics (data: Whole-Genome Alignments, SARS-CoV2)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/SARS_CoV2_alignments.png" width="400" /> 

	<font size=-2><u>Practical:</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/SARS-CoV2/practical.pdf" target="_blank">PDF</a>/<a href="https://thibaultleroyfr.github.io/courses.html" target="_blank">HTML</a> version. 
	[Left: Multiple SARS-CoV2 alignments]</font><br style="clear: both;" />
	</p>
		

		
	<h3>      -Detecting positive selection within genomes (data: Pool-seq data, European white oaks)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/wheat_circlize_10_10_cm.png" width="400" /> 
	Wheat, one of the most important food for humans, has been selectively bred across the globe for millennia. We recently explored the origins and patterns of genetic diversity in this hexaploid (AABBDD) species. <br />
	To do so, we used exome capture sequence data of a worldwide panel of 487 genotypes within the currently accessible wheat gene pool, including wild diploid and tetraploid relatives, domesticated tetraploid and hexaploid landraces, old cultivars and modern elite cultivars. <br />
	Comparing modern varieties (created since 1986) to founder material dating back to 1830 (i.e. landraces), we unveiled that modern wheat varieties showed an average loss of nucleotide diversity of 21.8% (33% on the subgenome A) suggesting that the last 2 centuries of breeding strongly reduced the within-species genetic diversity. <br />
	More broadly, we explored how 10,000 years of hybridization, selection, adaptation and plant breeding has shaped the genetic makeup of modern bread wheats. <br /><br />
	<font size=-2><u>More details:</u> Pont, C., Leroy, T., Seidel, M. et al. Tracing the ancestry of modern bread wheats. <a href="https://www.nature.com/articles/s41588-019-0393-z" target="_blank">Nature Genetics</a> 51,905â€“911. 
	[Left: plot created using the wonderful R package <a href="https://jokergoo.github.io/circlize_book/book/" target="_blank">Circlize</a>]</font><br style="clear: both;" />
	</p>
	
	<h2>Introduction to data analysis - Learning statistis with R (undergrad students)</h2>  
	
	<h2>Bioinformatics (all levels)</h2>  
		

</body>
</html>
