
<!DOCTYPE html PUBLIC>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="https://thibaultleroyfr.github.io/templates/templatetibomain.css" />
<title>Thibault Leroy - Genomics & Evolution</title>
</head>

<body>
    <div id="page">
        <div class="topNaviagationLink"><a href="index.html">Home</a></div>
        <div class="topNaviagationLink"><a href="courses.html">Courses</a></div>
        <div class="topNaviagationLink"><a href="publications.html">Publications & CV</a></div>
	    <div class="topNaviagationLink"><a href="https://github.com/ThibaultLeroyFr" target="_blank">Scripts & Pipelines</a></div>
	<div class="topNaviagationLink"><a href="contact.html">Contact</a></div>
	</div>
    <div id="mainPicture">
    	<div class="picture">
        	<div id="headerTitle">Thibault Leroy</div>
            <div id="headerSubtext">Courses</div>
        </div>
    </div>
        <div class="contentBox">
    	<div class="innerBox">

		
	<h1>Courses</h1>

	<h2>Evolutionary biology (2nd Master student, 2023-present, University of Toulouse, France)</h2>    
	<h3>- Mutations in animals and plants: an introduction (course only)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/SARS_CoV2_alignments_withbasecomp.png" width="400" /> 
	<br /><br />Main objective of the course: introducing recent empirical evidence about germline and somatic mutations in plants<br />
	<font size=-1><u>Course (2 hours):</u> <a href="https://thibaultleroyfr.github.io/courses/UEBioEvo_UPS_Mutation_201123.pdf" target="_blank"> PDF</a><br /><br /><br />
	</p>
		
	<h2>Population genomic approaches (Master & Phd students, 2020-2022, University of Vienna, Austria)</h2>      
		

	<h3>- Importance of preliminary analyses in population genomics (course + practical, data: Whole-Genome Alignments, SARS-CoV2)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/SARS_CoV2_alignments_withbasecomp.png" width="400" /> 
	Main objective of the course: introducing some good practices when we start to work on a new species/dataset in population genetics /genomics (first look at the referenge genome assembly, dubious sequences/individuals, regions of false positive variants, ...)<br />
	<font size=-1><u>Course:</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/SARS-CoV2/presentation_preliminary_analyses.pdf" target="_blank"> PDF</a><br />
	<font size=-1><u>Practical:</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/SARS-CoV2/practical.pdf" target="_blank"> PDF</a>/<a href="https://thibaultleroyfr.github.io/courses/proposal_sarscov2.html">HTML</a> version<br />
	<font size=-1><u>Tutorial:</u> <a href="https://thibaultleroyfr.github.io/courses/tutorial_sarscov2.html">HTML</a> version<br />
	<font size=-1><u>Data (practical):</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/SARS-CoV2/" target="_blank"> Files</a><br />
	<font size=-1><u>Last update:</u> April 2020<br />
	[Left: Multiple SARS-CoV2 alignments, including a plot that can be generated with the practical (after some efforts!) ]</font><br style="clear: both;" />
	</p>

	<h3>- Detecting positive selection within genomes (course only)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/genomic_footprint.png" width="400" /> 
	<br /><br /><br />
	Main objective of the course: brief introduction to different approaches: dN/dS, McDonald-Kreitman tests, nucleotide diversity & Tajima's D, detection of hard selective sweeps and Fst scans (master students)<br />
	<font size=-1><u>Course (2 hours):</u> <a href="https://thibaultleroyfr.github.io/courses/Leroy_GenApp_Footprints_selection.pdf" target="_blank"> PDF</a><br />
	<font size=-1><u>Last update:</u>  November 2020<br /><br /><br /><br /><br /><br />
	</p>
		
	<h3>- Detecting positive selection within genomes (course + practical, data: Pool-seq data, European white oaks)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/genomescans_oaks.png" width="400" />
	Main objective of the course: searching for footprints of natural (or artificial) selection by identifying SNPs exhibiting large allele frequency differences between populations (i.e. identifying SNPs with higher levels of differentiation than expected under strict neutrality).<br />
	<font size=-1><u>Course:</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/GenomeScans/presentation_GenomicApproaches_Genomescans.pdf" target="_blank"> PDF</a><br />
	<font size=-1><u>Practical:</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/GenomeScans/practical.pdf" target="_blank"> PDF</a>/<a href=" thibaultleroyfr.github.io/courses/proposal_footprintselection.html">HTML</a> version<br /> 
	<font size=-1><u>Data (practical):</u> <a href="https://github.com/ThibaultLeroyFr/Courses/blob/master/GenomicApproaches/GenomeScans/" target="_blank"> Files</a><br />
	<font size=-1><u>Last update:</u>  April 2020<br />
	[Left: Population differentiation XtX values for all SNPs locate in the first 30 Mb of the chromosome 1 of the oak genome. Outliers are shown in red, with the corresponding proportion per genomic window. The two plots can be generated with the practical (after some efforts!)]</font><br style="clear: both;" />
	</p>
	


	<h3>- Demographic inferences (course only)</h3> 
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/demographicinferences.png" width="400" />
	Main objective of the course: brief introduction of three different methods among the most popular ones to infer the demographic history of a given species (sequential Markovian coalescent, composite-likelihood based inferences and Approximate Bayesian Computation).<br />
	<font size=-1><u>Course (45 minutes):</u> <a href="https://github.com/ThibaultLeroyFr/thibaultleroyfr.github.io/blob/master/courses/Tibo_Workshop_Demographic_inferences.pdf" target="_blank"> PDF</a><br />
	<font size=-1>(last update:  December 2019)<br />
	<font size=-1><u>Course (2 hours):</u> <a href="https://github.com/ThibaultLeroyFr/thibaultleroyfr.github.io/blob/master/courses/Leroy_DemographicModelling2020.pdf" target="_blank"> PDF</a><br />
	<font size=-1>(last update:  October 2020)<br />
	[Left: Mutation, migration, and drift as evolutionary forces changing the site frequency spectrum]</font><br style="clear: both;" />
	</p>
		
	<h2>Introduction to data analysis - Learning statistis with R (undergrad students)</h2>  
	<br />
	<h2>Introduction to bioinformatics (all levels)</h2>  
	<br />
		

</body>
</html>
