
<!DOCTYPE html PUBLIC>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="templates/templatetibomain.css" />
<title>Thibault Leroy - Genomics & Evolution</title>
</head>

<body>
    <div id="page">
        <div class="topNaviagationLink"><a href="index.html">Home</a></div>
        <div class="topNaviagationLink"><a href="courses.html">Courses</a></div>
        <div class="topNaviagationLink"><a href="publications.html">Publications & CV</a></div>
	<div class="topNaviagationLink"><a href="https://github.com/ThibaultLeroyFr" target="_blank">Scripts & Pipelines</a></div>
	<div class="topNaviagationLink"><a href="contact.html">Contact</a></div>
	</div>
    <div id="mainPicture">
    	<div class="picture">
        	<div id="headerTitle">Thibault Leroy</div>
            <div id="headerSubtext">Evolutionary biologist</div>
        </div>
    </div>
        <div class="contentBox">
    	<div class="innerBox">

	<h1>In the spotlight</h1>
		
	<h2><b> The origin of heritable mutations in trees - PNAS (February 2024)</b><br /></h2>
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/TreeMutation_sampling_picture.png" width="400" />
	In this study, we explored the origin of new heritable mutations in plants, a question that was less explored compared to animals. While most animals have an early segregation of the germline, plant germline is generally assume to segregate very late in development, with a segregation typically assumed to occur near flowering in angiosperms. Such a late segregation is expected to allow the transmission of accumulated somatic mutations to progeny. In this study, we resequenced DNA from branches differentially exposed to sunlight in two tropical tree species, <i>Dicorynia guianensis</i> and <i>Sextonia rubra</i>. Contrary to expectations, we neither found more mutations in more exposed branches, nor any UV-specific footprint (mutation spectra). In addition, reconstructed phylogenies based on the somatic mutations do not follow branching patterns. Our study however unrevealed a large number of mutations, most of them observed at very low frequency in tissues. Despite their low allele frequency, we found that they can be transmitted to the next generation. Overall, our study challenges some common assumptions about somatic mutations in plants, revealing that low-frequency mutations can be heritable, mutation phylogenies deviate from tree branching patterns, and mutation rates do not seem to be highly influenced by UV exposure. These findings provide new perspectives on heritable somatic mutations, emphasizing their crucial role in plant evolution and suggesting more intricate links between plant growth, aging, UV exposure, and mutation rates than previously thought. <br /><br />
	<font size=-2><u>More details:</u> Schmitt et al. Low-frequency somatic mutations are heritable in tropical trees <i>Dicorynia guianensis</i> and <i>Sextonia rubra</i> <a href="https://www.pnas.org/doi/abs/10.1073/pnas.2313312121" target="_blank">PNAS 2024</a><br />
	</p>	
		
	<h2><b> Conserved mutational processes in animals and plants - eLife (October 2023)</b><br /></h2>
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: right; margin: 0px 15px 15px 0px;" src="img/eLife_figure_modif.png" width="400" />
	In this mini-review, I highlight how conserved the mutational processes seem to be in animals and plants. In animals, the origin of mutations was initially assumed to occur predominantly due to DNA replication errors and therefore to accumulate linearly, as the product of the number of cell divisions and the mutation rate per cell division. Detailed investigations over the last decade have however shown that germline mutations accumulate with age, rather than with the number of cell divisions. As a parallel in plants, Satake and collaborators have recently sequenced and assembled the genomes of two hardwood species of the Dipterocarpaceae botanical family living in central Borneo, Indonesia. These two species have notable differences in growth rates, with Shorea leprosula growing more than three times faster than S. laevis. By investigating somatic mutations within the two tree species, they remarkably found considerable support for an accumulation of plant somatic mutations with absolute time, not with the number of cell division. Given the apparent conservation of the mutational processes in animals and plants, my overall message is that plant and animal research communities interested by mutation have much to gain from collaborating with one another in the future.	 <br /><br />
	<font size=-2><u>More details:</u> Leroy. Plants: Re-evaluating the driving force behind mutations <a href="https://elifesciences.org/articles/89706" target="_blank">eLife 2023</a><br />
	</p>	

	<h2><b>"The dark side of the honeymoon": retracing rose breeding during the 19th - Preprint (June 2023)</b><br /></h2>
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: left; margin: 0px 15px 15px 0px;" src="img/Leroy_bioRxivRose.png" width="400" /> 
	In this preprint, we retraced the history of rose breeding during the 19th century, the golden age for roses in Europe! We indeed collected large phenotypic and genetic data from 200+ roses from an ancient collection and explored changes. Roses represent an ideal model to reconstruct the past history of breeding because they are reproduced through vegetative culture, allowing the maintenance of ancient varieties. This allows the genotyping and phenotyping of roses that were bred at different periods of time. We studied ~30 traits, including some presumably crucial for rose breeding: the duration of the blooming season, the color and number of petals and the floral scent (with volatile compounds quantified using a GC-mass spectrometry strategy). We have particularly studied wthe population structure, kinship and footprints of artificial selection, identifying genomic regions targeted by breeders. Finally, we performed a large GWA study considering both the genotypes and the phenotypes of the 200+ roses, generating the largest GWAS catalog for roses to date (<a href="https://roseGWASbrowser.github.io" target="_blank">roseGWASbrowser</a>). Our main take home message is associated with the crucial importance of preserving botanical and ancient rose collections to safeguard diversity and ensure a sustainable breeding for the long-term. This requires an increasing effort, especially regarding state funding. <br /><br />
	<font size=-2><u>More details:</u> Leroy et al. Dark side of the honeymoon: reconstructing the Asian x European rose breeding history through the lens of genomics <a href="https://www.biorxiv.org/content/10.1101/2023.06.22.546162v1" target="_blank">BioRxiv 2023</a><br />
	</p>
		
	<h2><b>Investigating genomic landscapes as a progression through time - Molecular Ecology (June 2023)</b><br /></h2>		
	<p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: right; margin: 0px 15px 15px 0px;" src="img/molecolhuiying.png" width="400" />
	Using a large population-based whole-genome sequencing effort (8 Populus species, >200 individuals in total), we investigated genomic landscapes of nucleotide diversity and divergence. Thanks to this multispecies sampling, we were able to investigate the genomic landscapes as progression through time considering all the species pairs along the Populus divergence gradient. We observed relatively conserved patterns of genomic divergence across the species pairs and found support for predominant signatures of linked selection, independently of the stage across the divergence gradient. Our results are also consistent with additional contribution of gene flow and standing genetic variation in shaping these genomic landscapes. Our study highlights the importance of investigating genomic patterns on multiple species across a divergence gradient to learn more about speciation processes. Remarkable work by Huiying Shang (PhD initially supervised by Christian Lexer, and then by Ovidiu Paun and I). The paper is dedicated to the memory of Christian. <br /><br />
	<font size=-2><u>More details:</u> Shang et al. Drivers of genomic landscapes of differentiation across a Populus divergence gradient <a href="https://onlinelibrary.wiley.com/doi/epdf/10.1111/mec.17034" target="_blank">Molecular Ecology 2023</a><br />
	</p>	

	
	
	<h1>About me</h1>
          <div class="contentText">
        <p style="margin: 10px 15px 15px 10px;font-size:16px;line-height:20px;"><img style="float: right; margin: 0px 15px 15px 0px;" src="img/logo_beeGEES_INRAE_GenPhyse_white.png" width="200" />
	I am a population geneticist, employed as permanent researcher (CRCN) at INRAE, Toulouse, France. My research is focused on the study of evolutionary forces: gene flow, drift, mutation and selection and their relative importance in divergence and speciation. My main model is honey bees but I continue to work on some other models through collaborations, including some models I studied over the last decade (plants: oaks, cereals, Populus, Bromeliads etc; animals: passerine birds, salmon etc). My interests also include genomics - in its broader sense - and statistical methods for population genetics. <br />
        <p><a href="https://evolgenomics.univie.ac.at/about-us/group-members/thibault-leroy/" target="_blank">More details here</a></div><br /></p>
		
        <div class="contentText"><br />
	<p><strong>Contact:<br /> </strong> Thibault (Tibo) Leroy - 
	<a href="mailto:thibault.leroy@inrae.fr?Subject=[Webpage]" target="_top">Email</a><br />
	Genetics, Physiology and Livestock Systems (GenPhySE)<br />
	Office: C406, INRAE Occitanie-Toulouse<br />
	24 Chemin de Borde Rouge, 31326 Castanet-Tolosan, France<br />
	<a rel="me" href="https://twitter.com/TiBoLeroyInEn">Twitter</a></p>
	<a rel="me" href="https://ecoevo.social/@TiBoLeroyInEn">Mastodon</a></p>

        </div>

</body>
</html>
